### ISSUE TITLE
Unable to access emails via API

### ISSUE THREAD

**Issue by chrisspen** [O365/python-o365#520]
The library appears to allow authentication, but nothing else.

Following the docs, I registered an app and then executed:

    from O365 import Account

    credentials = ('xxxxxxxxx', 'xxxxxxxxxxxxx')

    account = Account(credentials, auth_flow_type='credentials')
    if account.authenticate():
        print('Authenticated!')

which reported success. Great. So then I tried to get all messages in my inbox by modifying this code to:

    from O365 import Account, MSGraphProtocol

    credentials = ('xxxxxxxxxxxxxxxxx', 'xxxxxxxxxxxxxxxxx')
    tenant_id = 'xxxxxxxxxxxxxxxxxxxxxxxxx'

    protocol = MSGraphProtocol(default_resource='myusername@mydomain.onmicrosoft.com')

    account = Account(credentials, auth_flow_type='credentials', tenant_id=tenant_id, protocol=protocol)
    if account.authenticate():
        print('Authenticated!')

        messages = account.mailbox().get_messages()
        print('messages:', messages)

However, when I re-run this, I get the error:

    requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://graph.microsoft.com/v1.0/users/myusername@mydomain.onmicrosoft.com/messages?%24top=25 | Error Message: The token contains no permissions, or permissions can not be understood.

What's going on here? My app has all permissions for everything under my account. It just authenticated successfully. But now the token isn't being accepted?

I also tried removing the protocol, but then I get the error:

    requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://graph.microsoft.com/v1.0/messages?%24top=25 | Error Message: Resource not found for the segment 'messages'.

**Comment by janscas** [comment]
> The library appears to allow authentication, but nothing else.

The library does what it says it does. Another thing would be what you have achieved.

If you read the readme carefully you will notice you have to pass to either the account init or the authenticate method a list of permission scopes.

**Comment by janscas** [comment]
You will need to remove the token and reauthenticate requesting the needed scopes.

**Comment by chrisspen** [comment]
How do you remove the token?

**Comment by janscas** [comment]
Delete it. Its a text file.

**Comment by chrisspen** [comment]
>If you read the readme carefully you will notice you have to pass to either the account init or the authenticate method a list of permission scopes.

I guess this is what's confusing me. Most of the examples don't pass in a list of scopes. I've already given the app permission to everything in the Azure config page. I also have to re-specify all these permissions in the client when I request a token?

**Comment by chrisspen** [comment]
I tried deleting the token file and and then running this code:

    from O365 import Account, MSOffice365Protocol

    credentials = ('xxxxxxxxxxxxxx', 'xxxxxxxxxxxxxxxxxxxx')
    tenant_id = 'xxxxxxxxxxxxxxxxxxxxxxxxxx'

    protocol_office = MSOffice365Protocol()
    scopes_office = protocol_office.get_scopes_for('message all')

    account = Account(credentials, auth_flow_type='credentials', tenant_id=tenant_id, scopes=scopes_office)
    if account.authenticate():
        print('Authenticated!')

        mailbox = account.mailbox()
        inbox = mailbox.inbox_folder()

        sent_folder = mailbox.sent_folder()
        for message in sent_folder.get_messages():
            print(message)

but it still gives the error:

    Unable to fetch auth token. Error: (invalid_scope) AADSTS70011: The provided request must include a 'scope' input parameter. The provided value for the input parameter 'scope' is not valid. The scope message all is not valid.

**Comment by chrisspen** [comment]
I think I've figured it out. The problem wasn't the lack of scopes. The problem was I had granted my app delegated permissions, when I needed to give it application permissions and mark those all as having admin consent. I didn't notice the buttons in the webpage at first, mistaking them for column headers. Clicking the "application" button showed all the permissions I needed to grant.

Now I'm able to access email.
